scylla-cdc-lib/src/test/java/com/scylladb/cdc/lib/AlterTableIT.java

```diff
@@ -167,4 +178,193 @@ private synchronized void printDetails(RawChange change) {
     System.out.println(change.getAsObject("column1"));
     System.out.flush();
   }
+
+  @Test
+  public void alterTableBeforeNextPage() {
+    // A variation that alters the table right before the next page is fetched.
+    String keyspace = "testks2";
+    String table = "alternextpage2";
+    Session session;
+    int confidenceWindowSeconds = 15;
+    try (Cluster cluster = Cluster.builder().addContactPoint(hostname).withPort(port).build()) {
+      session = cluster.connect();
+      session.execute(
+          String.format(
+              "CREATE KEYSPACE IF NOT EXISTS %s WITH replication = {'class': 'SimpleStrategy', "
+                  + "'replication_factor': 1};",
+              keyspace));
+      session.execute(String.format("DROP TABLE IF EXISTS %s.%s;", keyspace, table));
+      session.execute(
+          String.format(
+              "CREATE TABLE IF NOT EXISTS %s.%s (column1 int, column2 int, column3 int, PRIMARY"
+                  + " KEY (column1, column2)) WITH cdc = {'enabled': 'true'};",
+              keyspace, table));
+
+      AtomicInteger counter = new AtomicInteger(0);
+      AtomicBoolean taskShouldStop = new AtomicBoolean(false);
+      // Start continuously populating the base table in the background
+      Runnable task =
+          () -> {
+            PreparedStatement ps =
+                session.prepare(
+                    String.format(
+                        "INSERT INTO %s.%s (column1, column2, column3) VALUES (?, ?, ?);",
+                        keyspace, table));
+            while (!taskShouldStop.get()) {
+              int current = counter.incrementAndGet();
+              // By limiting ourselves to 1 primary key we also ensure
+              // that the number of streams handled at a time will also be 1.
+              // This in turn should limit the number of library Tasks run on executor.
+              session.execute(ps.bind(1, 1, current));
+              Uninterruptibles.sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
+            }
+          };
+      Thread thread = new Thread(task);
+      thread.start();
+
+      AtomicBoolean failedDueToInvalidTypeEx = new AtomicBoolean(false);
+      AtomicBoolean unlockedNextPageLock = new AtomicBoolean(false);
+      AtomicBoolean consumedChangeAfterAlter = new AtomicBoolean(false);
+      AtomicLong lastConsumedTime = new AtomicLong(0);
+      AtomicReference<RawChange> firstChange, firstChangeAfterFetchingNewPage;
+      firstChange = new AtomicReference<>(null);
```

Comment:
```suggestion
      changeBeforeAlter = new AtomicReference<>(null);
```

Reply:
New version has `rawChangesBeforeAlter` and `rawChangesAfterAlter` lists